pkg/db/topology.go

```diff
@@ -169,6 +169,18 @@ func PersistComponent(component *pkg.Component) ([]uuid.UUID, error) {
 	return persistedComponents, tx.Error
 }
 
+func UpdateStatusAndSummarForComponent(id uuid.UUID, status string, summary v1.Summary) error {
+	component := &pkg.Component{}
+	tx := Gorm.Where("id = ?", id).Find(component)
```

Comment:
I think we can merge these 2 queries into a single one. 

ie.. `UPDATE component set status = :status, summary = :summary WHERE id = :id and status != :status and summary != summary`

This saves a round trip to the database, and 2 marshal/unmarshall/serialize steps. 
The return value of the update statement (no of rows) can then be used as a metric to track summary updates per minute.